adaptive_memory_v4.0.py

```
@@ -0,0 +1,1982 @@
+from typing import (
+    Any,
+    Awaitable,
+    Callable,
+    Dict,
+    List,
+    Literal,
+    Optional,
+    Union,
+    Set,
+    Tuple,
+)
```

Ruff (0.14.10)

Import from `collections.abc` instead: `Awaitable`, `Callable`

Import from `collections.abc`

(UP035)